src/clients/victoriaMetrics/promAgent.ts

```diff
@@ -0,0 +1,92 @@
+import promClient from 'prom-client';
+import assert from 'assert';
+import { sendToDiscordAdmins } from '../discord';
+
+const VM_IMPORT_PROMETHEUS_URL =
+  'http://localhost:8428/api/v1/import/prometheus';
+
+const MAX_PUSH_METRICS_ERRORS = 5;
+const PUSH_METRICS_ERRORS_RATE_LIMIT = 60 * 60 * 1000; // 1 hour
+
+export class PromAgent {
+  private metricNames: Set<string>;
+  private pushMetricsErrorsTimestamps: number[] = [];
+
+  constructor() {
+    this.metricNames = new Set<string>();
+  }
+
+  public addMetricName(metricName: string) {
+    assert(
+      !this.metricNames.has(metricName),
+      `Metric name already exists: ${metricName}`,
+    );
+
+    this.metricNames.add(metricName);
+  }
+
+  // TODO-kev: Below is purely for testing. Keep or remove?
+  public getMetricNames(): Set<string> {
+    return this.metricNames;
+  }
+
+  public async pushMetricsToVictoriaMetrics() {
+    const metrics = await promClient.register.metrics(); // Will call any collect() functions for gauges
+
+    const response = await fetch(VM_IMPORT_PROMETHEUS_URL, {
+      method: 'POST',
+      body: metrics,
+      headers: {
+        'Content-Type': 'text/plain',
+      },
+    });
+
+    if (!response.ok) {
+      // TODO-kev: Remove below
+      console.log('ERROR PUSHING METRICS');
+
+      // Alert while errors are less than MAX_PUSH_METRICS_ERRORS
+      const now = Date.now();
+      this.pushMetricsErrorsTimestamps.push(now);
```

Comment:
While the first entry is old, remove the first entry. Then check the size. Suppress if >5 or something. If alerting, push the timestamp up.